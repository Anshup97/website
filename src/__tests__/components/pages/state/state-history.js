import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import sampleScreenshots from 'screenshots'
import sampleHistory from 'state-history'
import StateHistory from '~components/pages/state/state-history'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    v1Json: {
      buildTime: '2020-06-02T10:16:47.239Z',
    },
    allContentfulDataDefinition: {
      nodes: [{ name: 'Date', fieldName: 'date' }],
    },
  }))
})

describe('Components : Pages : State : State history', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <StateHistory
          history={sampleHistory}
          screenshots={sampleScreenshots}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
