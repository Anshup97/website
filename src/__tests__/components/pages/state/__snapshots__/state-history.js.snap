// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : State : State history renders correctly 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          className="alignLeft"
          scope="col"
        >
          <span
            role="text"
          >
            Date
          </span>
        </th>
        <th
          className=""
          scope="col"
        >
          <span
            role="text"
          >
            New Tests
          </span>
        </th>
        <th
          className=""
          scope="col"
        >
          <span
            role="text"
          >
            Cases
          </span>
        </th>
        <th
          className=""
          scope="col"
        >
          <span
            role="text"
          >
            Negative
          </span>
        </th>
        <th
          className=""
          scope="col"
        >
          <span
            role="text"
          >
            Pending
          </span>
        </th>
        <th
          className=""
          scope="col"
        >
          <span
            role="text"
          >
            Hospitalized
          </span>
        </th>
        <th
          className=""
          scope="col"
        >
          <span
            role="text"
          >
            Deaths
          </span>
        </th>
        <th
          className=""
          scope="col"
        >
          <span
            role="text"
          />
        </th>
      </tr>
    </thead>
    <tbody
      className="state-history-table history"
    >
      <tr>
        <td
          className="alignLeft"
        >
          Fri Apr 17 2020
        </td>
        <td
          className=""
        >
          1,457
        </td>
        <td
          className=""
        >
          4,530
        </td>
        <td
          className=""
        >
          33,318
        </td>
        <td
          className=""
        >
          N/A
        </td>
        <td
          className=""
        >
          594
        </td>
        <td
          className=""
        >
          144
        </td>
        <td
          className=""
        >
          37,848
        </td>
      </tr>
      <tr>
        <td
          className="alignLeft"
        >
          Thu Apr 16 2020
        </td>
        <td
          className=""
        >
          2,314
        </td>
        <td
          className=""
        >
          4,345
        </td>
        <td
          className=""
        >
          32,046
        </td>
        <td
          className=""
        >
          N/A
        </td>
        <td
          className=""
        >
          553
        </td>
        <td
          className=""
        >
          133
        </td>
        <td
          className=""
        >
          36,391
        </td>
      </tr>
      <tr>
        <td
          className="alignLeft"
        >
          Wed Apr 15 2020
        </td>
        <td
          className=""
        >
          960
        </td>
        <td
          className=""
        >
          4,113
        </td>
        <td
          className=""
        >
          29,964
        </td>
        <td
          className=""
        >
          N/A
        </td>
        <td
          className=""
        >
          525
        </td>
        <td
          className=""
        >
          121
        </td>
        <td
          className=""
        >
          34,077
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
